platform/ios/src/MGLConfig.m

```diff
@@ -0,0 +1,38 @@
+#import "MGLConfig.h"
+
+static const CLLocationDistance MGLConfigHibernationRadiusDefault = 300.0;
+static const CLLocationDistance MGLConfigHibernationRadiusWide = 1200.0;
+
+NSString *const MGLMapboxMetricsProfile = @"MGLMapboxMetricsProfile";
+
+static NSString *const MGLConfigDefaultKey = @"Default";
+static NSString *const MGLConfigHibernationRadiusWideKey = @"WideGeoFence";
+
+@implementation MGLConfig
+
+- (instancetype) init {
+    self = [super init];
+    if (self) {
+        _MGLLocationManagerHibernationRadius = MGLConfigHibernationRadiusDefault;
+    }
+    return self;
+}
+
++ (nullable instancetype)sharedConfig {
+    static dispatch_once_t onceToken;
+    static MGLConfig *_sharedConfig;
+    dispatch_once(&onceToken, ^{
+        _sharedConfig = [[self alloc] init];
+    });
+    return _sharedConfig;
+}
+
+- (void)configurationFromKey:(NSString *)key {
+    if ([key isEqualToString:MGLConfigDefaultKey]) {
```

Comment:
As noted below, the call to `configurationFromKey:` will often happen when no value has been set for  `MGLMapboxMetricsProfile`. So, neither the default or "wide" value will be set here (normally). `_MGLLocationManagerHibernationRadius` will still be equal `300.0` since it is initialized as such in this class.

You could establish a default value for `MGLMapboxMetricsProfile` like we do for other known/expected user defaults (ref https://github.com/mapbox/mapbox-gl-native/blob/rclee-radius-config-agua/platform/ios/src/MGLMapboxEvents.m#L144-L148). However, I think that you could also just remove this extra default logic here since it is not going to be called anyway unless app developers set `MGLMapboxMetricsProfile` to `"Default"` but we are not going to ask anyone to do that.